### ISSUE TITLE
`Type alias circularly references` and `Index signature is missing` conflict

### ISSUE THREAD

**Issue by tqma113** [Lucifier129/farrow#76]
![image](https://user-images.githubusercontent.com/22792671/118652046-c217d600-b818-11eb-8d79-b1e51f219e44.png)

https://www.typescriptlang.org/play?#code/C4TwDgpgBAUgzgewHYBVzQLwCgpQD5RICuAtgEYQBOO+UcwlAlkgOY0FkIIA2EAhknaEi3bkKJIAJhABmzCJKHxkaSAG0AukIDeNXLmAIYAZQDyAOQAUASgBcdBszb6oAXyEAlCAGMElSQA89EysADSwiKjoAHxYWKCQESroAIJQGFBCxORUQsFOQpw8-IK4BMSi4lKy8oplSVGQKZo6evqGJhY29vmsbe71ui5qANYQID2OrBr2yo0QKTTuWN7cfHBwUABiXFDay7zAUDK7GftQ69tccTIS3sCMyFBkfJSWzGBEwLORqgvWe2WLzeJwQ1iAA

https://www.typescriptlang.org/play?#code/C4TwDgpgBAYg9nKBeKBvAvgKAJYDtgQBOAZgIYDG0AQqYWlpqJFAILKwJQQAeBuAJgGcoNOgH4owQgFdoALihkANoOgB6NZJkRG4au1FdeEAcPiIJU2VAXLVUDVtmZdzAMLtzRvkKgAlCHI4Qn4AHkEpPABzABooaVwAa1w4AHdcAD4oS20bRVIVdU0rHSZoABEDWm8TXwCgkPDI3Fj4pJT0rJzrWwL7RzsdV2gAUXZ64LCIwmi4hOS0zJrTDgsneXzCh2LtYdhxwMmmmZa59sWsnh9hQ26Nwe31l0dUgAsQEWqr2uEJxunZm0Fp0oNhhIMxEA

can't create perfect circular object type

https://github.com/microsoft/TypeScript/issues/14174

**Comment by Lucifier129** [comment]
Yes, the expected value of the JsonType type is generally the plain javascript object/value, not the class instance

If it is a class instance, it is expected to have a toJSON method that will be called during JSON.stringify to get the plain value, such as Date
